### PR TITLE
migrate CI to GitHub Actions and ensure recommended Elixir x OTP pairs are covered

### PR DESCRIPTION
**Pull request by adrianomitre**
This pull request migrate the CI from Travis CI to GitHub Actions, as requested by José Valim in [this comment](https://github.com/elixir-plug/plug/pull/957#issuecomment-647738665).

In addition, it makes sure all recommended Elixir and Erlang/OTP combinations are run on CI.

Quoting José Valim:
> I would still advise folks to go with:
> * For each Elixir version, add a run with earliest supported Erlang/OTP
> * For the last Elixir version, also test the latest supported Erlang/OTP

Please refer to the original discussion on https://github.com/dashbitco/broadway/pull/188 for the full rationale.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      - run: mix format --check-formatted
+        if: matrix.check_formatted
+      - name: Install Dependencies
+        run: |
+          mix local.hex --force
+          mix local.rebar --force
+          mix deps.get --only test
+      - run: mix compile --warnings-as-errors
+        env:
+          MIX_ENV: test
```
Comment (josevalim) [inline .github/workflows/ci.yml:44]: Honestly, all of the commands can run in MIX_ENV=test. Can we set it for the whole step somehow?

Comment (adrianomitre) [reply .github/workflows/ci.yml]: You probably mean "the whole job", right?

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

File: .github/workflows/ci.yml (hunk 1 above)
Comment (josevalim) [reply .github/workflows/ci.yml]: I think so. :)

**Pull request merged**